### DISCUSSION TITLE
Favicon and Title Not Displaying Correctly on Multiple Browsers

### DISCUSSION THREAD

**Discussion by o12486vs2** [zauberzeug/nicegui#1444] [Q&A]
### Question

Hello everyone,

I hope you're all doing well. I've encountered an issue with my NiceGUI application and I'm hoping someone might have experienced something similar or could provide some insight.

Here's the code I'm running:

```python
from pages.test import test
from nicegui import app, ui
import os
import uuid

app.add_static_files('/assets', 'assets')
app.add_static_files('/func', 'func')

@ui.page('/')
def startPage(): 
    test.codeHighlight()

from assets.logo import base64Logo
port = int(os.environ.get("PORT", 8080))  
ui.run(favicon=base64Logo.mainlogoFavicon, title='PIECODE', port=port, storage_secret=f'piecode_{uuid.uuid4()}', 
       uvicorn_logging_level='info')
```
When I execute this code and access the application from my primary browser, the favicon displays correctly. However, when I open the application in a different browser, both the favicon and the title do not show up. This same behavior occurs when I deploy the app to Heroku.

Any advice or solutions for this issue would be greatly appreciated.

Thank you in advance!

**Comment by rodja** [comment]
What is your primary browser? Which version? And on what kind of browsers is it not working? Are there any Java Script errors logged if it's not working? Could you share the content of `<head>` from the loaded html page?

**Reply by o12486vs2** [reply]
I'm not a developer, but mainly use Python for data analysis. While searching for a sleek web GUI, I was captivated by the charm of NiceGUI. Its efficient and convenient approach made me want to create content that allows me to build apps anywhere on the web using Python. During this process, I tried deploying to Heroku for the first time. At first, I assumed it was due to my lack of understanding about servers, but then I realized the same problem occurs locally as well. I even linked my test app to the domain [www.piecode.app](http://www.piecode.app/), wondering if it might be a domain-related issue, but the problem persisted, prompting me to reach out.

My primary browser is Chrome, but the issue is consistent across Microsoft Edge, Naver's Whale, and others. The problem persists on every browser, and no JavaScript errors are being logged in the console. The <head> section of the loaded HTML is quite extensive, but one thing I'm certain about is that the title and favicon I specified in ui.run() are not being loaded. Instead, only the default NiceGUI favicon and title are displayed.

**Reply by rodja** [reply]
Does it work with Firefox?

**Reply by rodja** [reply]
And what happens if you clear the cache?

**Reply by o12486vs2** [reply]
Even after clearing the cache, the issue persists.

![image](https://github.com/zauberzeug/nicegui/assets/76493297/67b2e87b-45a8-49b2-af26-f987d8aea52b)

**Comment by rodja** [comment]
It's very strange to see the correct title/favicon in one browser and not the other. These things are placed into the template before serving the page:

https://github.com/zauberzeug/nicegui/blob/57d9478540ea3231c794035cb6caa83176e7a82a/nicegui/templates/index.html#L2-L6

If I run 

```bash
 curl -s www.piecode.app | head -6
```

on your website it clearly states the default title/favicon. Are you sure it is running the code you shared in the original post?

**Reply by o12486vs2** [reply]
```py
from nicegui import app, ui
app.add_static_files('/assets', 'assets')
app.add_static_files('/func', 'func')
from pages.test import test
import os
import uuid

@ui.page('/')
def startPage(): 
    test.codeHighlight()

from assets.logo import base64Logo
port = int(os.environ.get("PORT", 8080))  
ui.run(favicon=base64Logo.mainlogoFavicon, title='PIECODE', port=port, storage_secret=f'piecode_{uuid.uuid4()}', 
       uvicorn_logging_level='info')
```

![캡처](https://github.com/zauberzeug/nicegui/assets/76493297/555fcbb6-a330-486d-9005-f4cc3a52e3c8)

**Comment by o12486vs2** [comment]
I apologize for the oversight. It seems that there was an issue with the app I created. When executing code written on the client side via WebSocket on the server, it appears to affect the entire project code. Fortunately, it's not an issue with nicegui. Thank you.
